### PR TITLE
PROJ-2483: Bottom Sheet Component

### PR DESCRIPTION
**Pull request by amireds**
Added a custom BottomSheet component to improve modal interactions. Integrated BottomSheet in the main App, adjusted the color and text high contrast logic, and modified conditional rendering logic. Additionally, updated type definitions for AppText and enhanced Storybook examples for better visualization.

### CODE REVIEW

File: packages/mobile/molecules/AppBSheet.tsx
Comment (kyeltech) [inline packages/mobile/molecules/AppBSheet.tsx:1]: @amireds  abeg move some of this into hook, too long

File: packages/mobile/molecules/AppBSheet.tsx [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+      throw new Error('Regular mode height must be a number')
+    }
+
+    return props as RegularProps
+  }
+}
+
+const AppBSheet = <T extends boolean>(props: AppBottomSheetProps<T>) => {
+  const checkedProps = checkBottomSheetProps(props)
+  // ref
```
Comment (nonsoo24) [inline packages/mobile/molecules/AppBSheet.tsx:130]: `AppBSheet` and `AppBottomSheetProps`.  I think the naming convention should be consistent.
E.g  `AppBSheet` and  `AppBSheetProps` or  `AppBottomSheet`  and `AppBottomSheetProps`

**Review by nonsoo24 — CHANGES_REQUESTED** [review]

File: packages/mobile/molecules/AppBSheet.tsx (hunk 1 above)
Comment (amireds) [reply packages/mobile/molecules/AppBSheet.tsx:130]: Okay,
I'll do that

File: packages/mobile/molecules/AppBSheet.tsx
Comment (amireds) [reply packages/mobile/molecules/AppBSheet.tsx:1]: Done

File: packages/mobile/molecules/AppBSheet.tsx (hunk 1 above)
Comment (amireds) [reply packages/mobile/molecules/AppBSheet.tsx]: Done

**Pull request merged**